front/src/modules/simulationResult/components/SpaceTimeChart/useLazyProjectTrains.ts

```diff
@@ -21,27 +27,44 @@ const useLazyProjectTrains = ({
   path,
 }: UseLazyProjectTrainsOptions) => {
   const dispatch = useAppDispatch();
-  const loaderRef = useRef<TrainProjectionLazyLoader>();
+  const loaderRef = useRef<TrainTrackProjectionLazyLoader>();
   const timetableItemsByIdRef = useRef<Map<TimetableItemId, TimetableItem>>(new Map());
   const [projectedTrainsById, setProjectedTrainsById] = useState<
     Map<TimetableItemId, TrainSpaceTimeData>
   >(new Map());
+  const projectionType = useSelector(getProjectionType);
+  const timetableItemIdProjected = useSelector(getTrainIdUsedForProjection);
 
   useEffect(() => {
-    if (!path) return undefined;
+    if (!path || !projectionType) return undefined;
```

Comment:
You won't need this check if projectionType is always defined
```suggestion
    if (!path) return undefined;
```

Reply:
Done